`Plagiarism check`: Improve new plagiarism notification email

rriyaldhi opened the pull request:
<!-- Thanks for contributing to Artemis! Before you submit your pull request, please make sure to check the following boxes by putting an x in the [ ] (don't: [x ], [ x], do: [x]) -->
<!-- If your pull request is not ready for review yet, create a draft pull request! -->

### Checklist
#### General
<!-- You only need to choose one of the first two check items: Generally, test on the test servers. -->
<!-- If it's only a small change, testing it locally is acceptable and you may remove the first checkmark. If you are unsure, please test on the test servers. -->
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Server
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/) and ensured that the layout is responsive.
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context
<!-- Why is this change required? What problem does it solve? -->
<!-- If it fixes an open issue, please link to the issue here. -->

Closes [6170](https://github.com/ls1intum/Artemis/issues/6170).

### Description
<!-- Describe your changes in detail -->

- Modify the new plagiarism case notification body according to [6170](https://github.com/ls1intum/Artemis/issues/6170)
- Change aspoLink to [https://portal.mytum.de/archiv/kompendium_rechtsangelegenheiten/apso/Lesb-F-APSO-vom-18-03-2011-mit-3-AeS-vom-27-04-18-ohne-Markierungen.pdf/view](https://portal.mytum.de/archiv/kompendium_rechtsangelegenheiten/apso/Lesb-F-APSO-vom-18-03-2011-mit-3-AeS-vom-27-04-18-ohne-Markierungen.pdf/view)
- Modify the new plagiarism case notification body from GitHub markdown to HTML
- Move the greeting and farewell of new plagiarism case email to the notification content
- Remove email footer (unsubscribe text) from new plagiarism case email
- Fix text color Open Artemis Button in an email
- Add a "Reply to Accusation" button to the student plagiarism case detail page
- Add text that shows the deadline for replying to the accusation
- Modify the content column of the post table to varchar(5000)

### Steps for Testing
<!-- Please describe in detail how the reviewer can test your changes. -->

**For a detailed example of testing, please see this screencast:**
https://user-images.githubusercontent.com/10885503/216544931-dababba7-45e2-4d41-81ba-a9d9d2948fbc.mov

Prerequisites:
- Test Server 0
- 1 Course (e.g. [General Testing Course](https://artemis-test0.artemis.in.tum.de/courses/5/exercises))
- 1 Instructor (registered to the course) (e.g. artemis_test_user_1 or artemis_test_user_2)
- 3 Students (2 test accounts and 1 account with your email) (registered to the course)

1. Log in to Artemis as an Instructor
2. Navigate to the Course Management
3. Navigate to Exercises
4. Click Import Programming Exercise
5. Search **Test 6174**
6. Click Import of exercise **Test 6174**
7. Insert Title and Short Name
8. Set the release date to current time
9. Ensure that Allow Online Editor is checked
10. Click Import
11. Log in to Artemis as the first student
12. Select the Course
13. Click Start exercise of the newly created programming exercise
14. Click Open code editor, then click Open graded code editor
15. Click BubbleSort.java
16. Insert any code inside the performSort method
17. Click Submit
18. Log out and log in as the second student
19. Select the Course
20. Click Start exercise of the newly created programming exercise
21. Click Open code editor, then click Open graded code editor
22. Click BubbleSort.java
23. Insert bubble sort code inside the performSort method
24. Click Submit
25. Log out and log in as the user with your email
26. Select the Course
27. Click Start exercise of the newly created programming exercise
28. Click Open code editor, then click Open graded code editor
29. Click BubbleSort.java
30. Insert bubble sort code inside the performSort method
31. Click Submit
32. Log in to Artemis as an Instructor
33. Navigate the programming exercise that you created
34. Click Check Plagiarism
35. Click Detect plagiarism
36. Click the result, then click Confirm
37. Click the course name of the breadcrumb navigation
38. Click Plagiarism Cases
39. Scroll to the programming exercise that you created
40. Click the case of the account with your email
41. Click Notify student
42. Click Preview tab
43. Ensure that the content looks good in preview mode.
44. Click Save
45. Check your email and verify that the content is still the same as in step 43
46. Log in as the student
47. Navigate to the plagiarism case
48. Verify there is a red button with the text "Reply to accusation" and also a text that states the deadline to reply
49. Verify that the student can reply to the case

### Review Progress
<!-- Each Pull Request should be reviewed by at least two other developers. The code, the functionality (= manual test) and the exam mode need to be reviewed. -->
<!-- The reviewer or author check the following boxes depending on what was reviewed or tested. All boxes should be checked before merge. -->
<!-- You can add additional checkboxes if it makes sense to only review parts of the code or functionality. -->
<!-- When changes are pushed, uncheck the affected boxes. (Not all changes require full re-reviews.) -->
<!-- All PRs that might affect the exam mode (e.g. change a client component that is also used in the exam mode) need an additional verification that the exam mode still works. -->

#### Code Review
- [x] Review 1
- [ ] Review 2
#### Manual Tests
- [x] Test 1
- [ ] Test 2

### Test Coverage
<!-- Please add the test coverages for all changed files here. You can see this when executing the tests locally (see build.gradle and package.json) or when looking into the corresponding Bamboo build plan. -->
<!-- Lines are the main reference but a significantly lower branch percentage can indicate missing edge cases in the tests. -->
<!-- Note: You may use the table below or copy the file coverage from the Codecov bot's comment. -->
| Class/File | Branch | Line |
|------------|-------:|-----:|
| Posting.java | n/a | 90% |
| plagiarism-case-instructor-detail-view.component.ts | 88.39% | 98.9% |
| plagiarism-case-student-detail-view.component.ts | 76.74% | 94.87% |
| metis.util.ts | 100% | 100% |
| post.component.ts | 33.33% | 86.2% |
| posting-create-edit.directive.ts | 80% | 100% |
| post-footer.component.ts | 90.47% | 100% |
| post-footer.component.ts | 90.47% | 100% |

### Screenshots
**Instructor Plagiarism Case Detail Page**
![instructor](https://user-images.githubusercontent.com/10885503/216545369-6af5924d-af02-4290-9c51-9f22484ab832.png)

**Student Plagiarism Case Detail Page**
![student](https://user-images.githubusercontent.com/10885503/216545396-3a1268cc-bff6-47b3-979e-b1d9b22701bb.png)

**Email**
<img width="1175" alt="email" src="https://user-images.githubusercontent.com/10885503/216545417-b3a09738-c6fa-4f31-b825-cbe5b62d51dc.png">

rriyaldhi commented on src/main/resources/i18n/messages.properties:
```diff
@@ -95,7 +95,6 @@ email.notification.aux.difficulty.hard=Hard
 
 # Plagiarism
 email.notification.title.post.plagiarismVerdict=Verdict reached in plagiarism case for exercise {0}
-email.notification.title.post.plagiarismCase=New plagiarism case in exercise {0}
```
This is not used anymore. The one that is specified at "artemisApp.plagiarism.plagiarismCases.notification.title" is used instead.

ge65cer reviewed (changes requested):
I suggest just changing the link and not the body, as new errors appeared:
- The text is now too long for a post (max. 1000 chars) when exercise/course have a long title
- The greeting is missing in the post (Artemis notification)
- The consequence is not always a 'U' but could just be a warning
- Instead of Nextcloud, maybe this 'official' link? https://portal.mytum.de/archiv/kompendium_rechtsangelegenheiten/apso/Lesb-F-APSO-vom-18-03-2011-mit-3-AeS-vom-27-04-18-ohne-Markierungen.pdf/view

rriyaldhi commented on src/main/resources/templates/mail/notification/plagiarismCaseEmail.html:
```diff
@@ -5,20 +5,13 @@
-            )}">Notification Content for Plagiarism Case
-            </span>
-        <div class="bordered-content">
+        <div>
```
Greeting is part of the content

rriyaldhi commented on src/main/resources/templates/mail/notification/plagiarismCaseEmail.html:
```diff
@@ -5,20 +5,13 @@
         </div>
         <th:block th:replace="mail/notification/fragments :: openInArtemisButton"/>
     </div>
-    <th:block th:replace="mail/notification/fragments :: farewell"/>
```
Farewell is part of the content

rriyaldhi commented:
> I suggest just changing the link and not the body, as new errors appeared:
> 
> * The text is now too long for a post (max. 1000 chars) when exercise/course have a long title
> * The greeting is missing in the post (Artemis notification)
> * The consequence is not always a 'U' but could just be a warning
> * Instead of Nextcloud, maybe this 'official' link? https://portal.mytum.de/archiv/kompendium_rechtsangelegenheiten/apso/Lesb-F-APSO-vom-18-03-2011-mit-3-AeS-vom-27-04-18-ohne-Markierungen.pdf/view

Hi @ge65cer, I've adjusted the PR according to your suggestions. Thanks!

- The text is now 5000 chars max
- There is now greeting in the post
- Changed the notification content
- Changed the link.

ge65cer commented on src/main/java/de/tum/in/www1/artemis/domain/metis/Posting.java:
```diff
@@ -37,9 +37,8 @@ public abstract class Posting extends DomainObject {
-    @Size(max = 1000)
-    @Column(name = "content", length = 1000)
+    @Size(max = 5000)
+    @Column(name = "content", length = 5000)
```
@stefanwaldhauser Please check if this change is reasonable.

ge65cer commented on src/main/webapp/app/course/plagiarism-cases/student-view/detail-view/plagiarism-case-student-detail-view.component.html:
```diff
@@ -41,7 +41,23 @@ <h5>{{ 'artemisApp.plagiarism.plagiarismCases.verdict.warning' | artemisTranslat
+                                        | artemisTranslate : { deadline: posts[0].creationDate.add(7, 'day').format('DD.MM.YYYY') }
+                                }}
+                            </h5>
+                            <jhi-button [btnType]="ButtonType.ERROR" (click)="postComponent.openCreateAnswerPostModal()" title="artemisApp.plagiarism.replyToAccusation">
```
Type "Warning" instead of "Error" maybe.

rriyaldhi replied:
Changed to `ButtonType.WARNING`

ge65cer commented on src/main/resources/config/liquibase/changelog/20230201102700_changelog.xml:
```diff
@@ -0,0 +1,8 @@
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.18.xsd">
+    <changeSet author="rriyaldhi" id="20230201102700">
+        <modifyDataType tableName="post" columnName="content" newDataType="varchar(5000)"/>
```
@stefanwaldhauser Please re-check.

ge65cer commented on src/main/webapp/app/course/plagiarism-cases/student-view/detail-view/plagiarism-case-student-detail-view.component.scss:
```diff
@@ -15,4 +15,8 @@
     .case-body {
         padding: 1.5rem;
     }
+    .separation-line {
```
Can be removed.

rriyaldhi replied:
Removed

ge65cer commented on src/main/webapp/app/course/plagiarism-cases/student-view/detail-view/plagiarism-case-student-detail-view.component.html:
```diff
@@ -41,7 +41,23 @@ <h5>{{ 'artemisApp.plagiarism.plagiarismCases.verdict.warning' | artemisTranslat
+                            <jhi-button [btnType]="ButtonType.ERROR" (click)="postComponent.openCreateAnswerPostModal()" title="artemisApp.plagiarism.replyToAccusation">
+                            </jhi-button>
+                        </div>
+                        <div class="separation-line mt-3"></div>
```
```suggestion
                        <hr />
```

ge65cer commented on src/main/webapp/app/course/plagiarism-cases/student-view/detail-view/plagiarism-case-student-detail-view.component.html:
```diff
@@ -41,7 +41,23 @@ <h5>{{ 'artemisApp.plagiarism.plagiarismCases.verdict.warning' | artemisTranslat
+                            <jhi-button [btnType]="ButtonType.ERROR" (click)="postComponent.openCreateAnswerPostModal()" title="artemisApp.plagiarism.replyToAccusation">
+                            </jhi-button>
+                        </div>
+                        <div class="separation-line mt-3"></div>
```
Can be removed.

rriyaldhi replied:
Removed

ge65cer commented on src/main/webapp/i18n/de/plagiarism.json:
```diff
@@ -44,6 +44,8 @@
             "resetFilter": "Filter zurücksetzen",
             "loading": "Laden...",
             "fetchingResults": "Plagiate werden geladen...",
+            "replyToAccusation": "Antwort auf den Vorwurf",
```
```suggestion
            "replyToAccusation": "Stellungnahme zu dem Vorwurf",
```

ge65cer commented on src/main/webapp/i18n/de/plagiarism.json:
```diff
@@ -44,6 +44,8 @@
             "loading": "Laden...",
             "fetchingResults": "Plagiate werden geladen...",
+            "replyToAccusation": "Antwort auf den Vorwurf",
+            "replyToAccusationDeadlineNotification": "Sie haben bis zum {{ deadline }} Zeit, sich zu dem Vorwurf zu äußern",
```
```suggestion
            "replyToAccusationDeadlineNotification": "Du kannst dich spätestens bis zum <b>{{ deadline }}</b> zu Vorwurf zu äußern.",
```

ge65cer commented on src/main/webapp/i18n/en/plagiarism.json:
```diff
@@ -44,6 +44,8 @@
             "loading": "Loading...",
             "fetchingResults": "Fetching plagiarism results...",
+            "replyToAccusation": "Reply to Accusation",
+            "replyToAccusationDeadlineNotification": "You have until {{ deadline }} to reply to the accusation",
```
```suggestion
            "replyToAccusationDeadlineNotification": "You can provide a statement to this accusation until <b>{{ deadline }}</b> the latest.",
```

ge65cer reviewed (commented):
Tested and seems to work fine, only small changes. 👍

Santia-go reviewed (approved):
Tested on TS1 with a modeling exercise
Working as Expected! ✅
I received the email with the corresponding links to the mentioned document and to the plagiarism case to review the similarity and provide an answer. 

Approved! 👍🏻
